### PR TITLE
add some stress tests

### PR DESCRIPTION
**Pull request by MikeStall**
getting failures, but using PR to comment.

### CODE REVIEW

File: src/PAModel/CanvasDocument.cs [hunk 1]
```diff
@@ -309,6 +309,8 @@ internal CanvasDocument(CanvasDocument other)
             _logoFile = other._logoFile != null ? new FileEntry(other._logoFile) : null;
             _entropy = other._entropy.JsonClone();
             _checksum = other._checksum.JsonClone();
+
+            this._idRestorer = new UniqueIdRestorer(this._entropy);
```
Comment (MikeStall) [inline src/PAModel/CanvasDocument.cs:313]: This looks like a real bug. 
The regular load paths allocate this, was missing from ctor path, and causing null refs.

File: src/PAModel/MsAppTest.cs [hunk 2]
```diff
@@ -14,6 +16,126 @@ namespace Microsoft.PowerPlatform.Formulas.Tools
...
+            using (var temp2 = new TempFile())
+            {
+                doc1.SaveToMsApp(temp1.FullPath);
+                doc2.SaveToMsApp(temp2.FullPath);
+
+                var doc1NoEntropy = RemoveEntropy(temp1.FullPath);
+                var doc2NoEntropy = RemoveEntropy(temp2.FullPath);
+
+                Compare(doc1NoEntropy, doc2NoEntropy, Console.Out);
+            }
```
Comment (MikeStall) [inline src/PAModel/MsAppTest.cs:82]: Getting tons of errors from this test.

File: src/PAModel/MsAppTest.cs [hunk 3]
```diff
@@ -14,6 +16,126 @@ namespace Microsoft.PowerPlatform.Formulas.Tools
...
+                errors.ThrowOnErrors();
+
+                
+#if false
+                // $$$ We get different checksums when deleting in-memory vs. files.
+                // In some cases, files are just rearranged (\images),
+                // but in other cases, look different. 
+
+                // Reset Entropy on in-memory model
+                doc1._entropy = new Entropy();
```
Comment (MikeStall) [inline src/PAModel/MsAppTest.cs:115]: This constantly fails too. The expectation (perhaps wrongly?) is that clearing the entropy in memory would have same affect as deleting entropy on disk. 
In reality, lots of unexpected diffs.

File: src/PAModel/CanvasDocument.cs (hunk 1 above)
Comment (MikeStall) [reply src/PAModel/CanvasDocument.cs:313]: actually, I think this still has an issue, in Merger.ApplyDelta(), we delete entropy via setting to null. That means idRestorer would be pointing to an orphaned entropy object.

File: src/PAModel/Checksum/ChecksumMaker.cs [hunk 4]
```diff
@@ -299,6 +299,7 @@ private static void ChecksumJson(Context ctx, IHashMaker hash, JsonElement je)
                             }
                             else
                             {
+                                hash.AppendNull();
```
Comment (MikeStall) [inline src/PAModel/Checksum/ChecksumMaker.cs:302]: >hash.AppendNull(); [](http://example.com/codeflow?start=32&length=18)

Will this require a new hash C9?

Comment (lesaltzm) [reply src/PAModel/Checksum/ChecksumMaker.cs:302]: Yeah, but actually I think I can fix it without doing that. The bug is with DebugTextHashMaker, if you write an object like 
`{key: null, otherKey: 123}`, the json writer complains that you wrote two keys in a row.

**Comment by MikeStall** [comment]
Will submit more recent version.

**Pull request closed without merging**